src/tbb/task_group_context.cpp

```diff
@@ -47,6 +48,12 @@ void tbb_exception_ptr::throw_self() {
     std::rethrow_exception(my_ptr);
 }
 
+void tbb_exception_ptr::throw_self_and_destroy() {
+    auto temp_ptr = my_ptr;
+    destroy();
+    std::rethrow_exception(temp_ptr);
+}
```

Comment:
I would rename the method to `rethrow_and_destroy`.

Also take a look at `throw_self()` right above; there is a `fix_broken_rethrow` workaround which I believe should be used in the new method as well.

Finally, if `throw_self` is not used anymore, let's remove it.